include/seqan3/range/view/take.hpp

```diff
@@ -0,0 +1,508 @@
+        return {ranges::cend(urange)};
+    }
+
+    //!\copydoc end()
+    sentinel_type cend() const noexcept
+        requires const_iterable_concept<urng_t>
+    {
+        return {ranges::cend(urange)};
+    }
+    //!\}
+
+    /*!\brief Returns the number of elements in the view (only available for "exactly" specialisation!).
+     * \returns The number of elements in the view.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type size() const noexcept
+        requires exactly
+    {
+        return target_size;
+    }
+
+    /*!\brief Convert this view into a container implicitly.
+     * \tparam container_t Type of the container to convert to; must satisfy seqan3::container_concept and the
+     *                     seqan3::reference_t of both must model seqan3::common_reference_concept.
+     * \returns This view converted to container_t.
+     */
+    template <container_concept container_t>
+    operator container_t()
+    //!\cond
+        requires common_reference_concept<reference_t<container_t>, reference>
+    //!\endcond
+    {
+        container_t ret;
+        ranges::copy(begin(), end(), ranges::back_inserter(ret));
+        return ret;
+    }
+};
+
+//!\brief Template argument type deduction guide that strips references.
+//!\relates seqan3::detail::view_take
+template <typename urng_t,
+          bool exactly = false,
+          bool or_throw = false>
+view_take(urng_t, size_t) -> view_take<std::remove_reference_t<urng_t>, exactly, or_throw>;
+
+// ============================================================================
+//  take_fn (adaptor definition)
+// ============================================================================
+
+/*!\brief View adaptor definition for view::take and view::take_or_throw.
+ * \tparam or_throw Whether to throw an exception when the input is exhausted before the end of line is reached.
+ */
+template <bool exactly, bool or_throw>
+class take_fn : public pipable_adaptor_base<take_fn<exactly, or_throw>>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<take_fn<exactly, or_throw>>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       Call the view's constructor with the underlying view as argument.
+     * \returns     An instance of seqan3::detail::view_take.
+     */
+    template <view_concept urng_t>
+    static auto impl(urng_t urange, size_t const target_size)
+    {
+        return view_take<urng_t, exactly, or_throw>{std::move(urange), target_size};
+    }
+
+    /*!\brief       Call the view's constructor with the underlying range wrapped in seqan3::view::all as argument.
+     * \returns     An instance of seqan3::detail::view_take.
+     */
+    template <viewable_range_concept urng_t>
```

Comment:
Couldn't inherit this from the pipable_adaptor_base?

Reply:
Then we don't need to write this every time, which I think can be DRYed out.

Reply:
> Couldn't inherit this from the pipable_adaptor_base?

It does.

> Then we don't need to write this every time, which I think can be DRYed out.

The problem here is that that functor type is templatised and that we use that parameter for deducing the returned view. Moving the `view::all` up into `pipable_adaptor` would work, but would break other specialisations of `pipable_adaptor` like `view::deep`. 

I have a large TODO for refactoring the pipable adaptor on my mental list, because it currently can't fold multiple adaptors into one without applying to a range and I will revisit this topic than, as well. But it is not trivial and I am trying to focus on more urgent things right now.